tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/EncodingBase.scala

```diff
@@ -0,0 +1,6 @@
+package at.forsyte.apalache.tla.bmcmt
+
+trait EncodingBase {
+  protected val oopsla19EncodingType = "oopsla19"
+  protected val arraysEncodingType = "arrays"
+}
```

Comment:
Any particular reason why the fields are protected in a trait and not public in an object?
```suggestion
object EncodingBase {
  val oopsla19EncodingType = "oopsla19"
  val arraysEncodingType = "arrays"
}
```

Reply:
I imagined `EncodingBase` only as a means to inherit the encoding types, thus I used `trait`. Is there any advantage in using `object`?

Reply:
I mean, this allows access to the encoding type from anywhere. `oopsla19EncodingType` is essentially a global constant this way.

Reply:
I see, I just didn't think we would need this anywhere else.

Reply:
Committing this change as is will break the unit tests, I will do it in a separate commit.